src/wallet/spend.cpp

```diff
@@ -702,7 +704,8 @@ util::Result<SelectionResult> ChooseSelectionResult(interfaces::Chain& chain, co
     }
 
     // Deduct change weight because remaining Coin Selection algorithms can create change output
-    max_selection_weight -= (coin_selection_params.change_output_size * WITNESS_SCALE_FACTOR);
+    int change_outputs_weight = coin_selection_params.change_output_size * WITNESS_SCALE_FACTOR;
+    max_selection_weight = std::max((max_selection_weight - change_outputs_weight), 0);
```

Comment:
Same as above, would be nice to error out early if `max_selection_weight <= 0`. No need to run the selection algorithms under such circumstance.

Reply:
Fixed, and also add a test case for it.